load.ts

```diff
@@ -1,14 +1,18 @@
-import type { PSMap, PSModule, PSValue } from "./types.ts";
+import type { PSEnv, PSMap, PSModule, PSValue } from "./types.ts";
 import type { Operation } from "./deps.ts";
 
 import { expect, useAbortSignal } from "./deps.ts";
 import { exclude, lookup } from "./psmap.ts";
 import { createYSEnv, letdo, parse } from "./evaluate.ts";
 
-export function* load(
+export interface LoadOptions {
   location: string | URL,
-  base?: string,
-): Operation<PSModule> {
+  base?: string;
+  env?: PSEnv;
+}
+
+export function* load(options: LoadOptions): Operation<PSModule> {
```

Comment:
In order to support loading modules uniformly, we need to pass an eval environment.